### DISCUSSION TITLE
[Question] Is there a REST API to pull Dashboard and Slice Views

### DISCUSSION THREAD

**Discussion by sashank** [apache/superset#14205] [Q&A / Help]
We are looking for ways to integrate superset into our web application . IFrame is not an option for us (for reasons like security and user experience friction with re-authenticating inside iframe etc. ) and i read from other threads that JS API is still on roadmap but no definitive timelines. 

After some analysis , figured out that SuperSet is based on Flask App Builder , and FAB has a[ REST API](http://flask-appbuilder.readthedocs.io/en/latest/quickhowto.html#rest-api) to pull the views . 

So can anyone help us by answering whether it is trivial to use those API's and pull the Dashboard and Slice Views from Apache Superset ?

**Comment by sashank** [comment]
We figured out it s trivial to pull dashboards via REST API

**Comment by bipinsoniguavus** [comment]
@sashank can you please help me telling your findings why you concluded it to be trivial to pull the dashboards via REST API. Thanks

**Comment by sashank** [comment]
@bipinsoniguavus  , each dashboard and slice is accessible via unique URL , check the link embedded in above answer , you need to flex a bit for authentication layer though

**Comment by iamcoolkranthi** [comment]
HI , any documentation is there how to do .
i am new to python

**Comment by holgerbrandl** [comment]
Same here, to enable automatic builds that preconfigure some dashboards, it would be great to have an example about how to to so.

**Comment by onlyphantom** [comment]
Copying from @mistercrunch's answer on #4708 for anyone stumbled upon this thread. Answer is his:

> Flask App Builder (FAB) offers a CRUD REST API out of the box for all models.
> Check out:

```
/slicemodelview/api
/slicemodelview/api/read
/dashboardmodelview/api
/dashboardmodelview/api/read
```
> http://flask-appbuilder.readthedocs.io/en/latest/quickhowto.html#rest-api

**Comment by kalimuthu123** [comment]
@onlyphantom Actually my requirement is pass the query through an api for fetching the data is there any api available in superset

**Comment by veerappans** [comment]
@kalimuthu123 , did you find a solution for this problem. I am trying to do something similar.

**Comment by nytai** [comment]
The closest we have now is the chart/data api (used by charts for fetching data). It accepts json and returns data, docs available here:
https://superset.apache.org/docs/rest-api

**Comment by veerappans** [comment]
thanks @nytai. I found these documents too. But I am not aware of how to construct a superset query for request body. Is there a way to use slice id?

**Comment by suddjian** [comment]
I suggest you check out the docs at `[superset-url]/swagger/v1`. The api endpoints for dashboards, charts, and the query data api can all be found there, as well as many other api endpoints. Note that you may need to run superset locally in order to access the docs page.
